src/alpaca_parser.yrl

```diff
@@ -449,11 +452,20 @@ tuple -> '(' tuple_list ')' :
 
 infix -> term op term : make_infix('$2', '$1', '$3').
 
+literal_fun -> fn terms '->' simple_expr:
```

Comment:
Probably a good idea, yes.  Just have to break down terms, etc a little better I expect :)